### DISCUSSION TITLE
How to use language translations/labels in MD files?

### DISCUSSION THREAD

**Discussion by MiguelVis** [datenstrom/community#689] [Ask a question] [answered]
Hi all,

Is there any way to include a translated label directly into a MD file? - ie, something like:

```
[MenuTitle]
- [MenuOptionIntroduction]
- [MenuOptionSupport]
```

That would help with multi language sites (no need to edit the same file in every language in some cases).

The language code should be given from the file - ie:

```
---
Title: Über
Language: de
---
Wo zusammenwächst was zusammen gehört.
```

or from the multi language contents structure - ie:

```
├── content               
│   ├── 1-en              
│   │   ├── 1-home        = http://website/
│   │   ├── 9-about
│   │   └── shared    
│   ├── 2-de              
│   │   ├── 1-home        = http://website/de/
│   │   ├── 9-about
│   │   └── shared    
│   └── 3-sv              
│       ├── 1-home        = http://website/sv/
│       ├── 9-about
│       └── shared    
├── media                 
└── system
```

**Comment by markseuffert** [comment, marked as answer]
You can use the API to [access language settings ](https://datenstrom.se/yellow/help/api-for-developers#yellow-language) in your HTML layout files. Here's an example layout:

``` php
<?php $this->yellow->layout("header") ?>
<div class="content">
<div class="main" role="main">
<h1><?php echo $this->yellow->page->getHtml("titleContent") ?></h1>
<ul>
<li><?php echo $this->yellow->language->getTextHtml("menuOptionIntroduction") ?></li>
<li><?php echo $this->yellow->language->getTextHtml("menuOptionSupport") ?></li> 
</ul>
<?php echo $this->yellow->page->getContent() ?>
</div>
</div>
<?php $this->yellow->layout("footer") ?>
```

As a developer you can also write an extension to use language settings in Markdown content files. For example an [event handler](https://datenstrom.se/yellow/help/api-for-developers#events) for `onParseContentShortcut()` that parses the input and returns the requested label(s) as output. Hope this helps, let me know if you need more API examples.

**Reply by MiguelVis** [reply]
Thanks @markseu, I will look at both possibilities.

**Reply by GiovanniSalmeri** [reply]
As suggested by Mark, a tiny extension like this would do the job:

```
<?php
// Text extension

class YellowText {
    const VERSION = "0.8.15";
    public $yellow;         // access to API
    
    // Handle initialisation
    public function onLoad($yellow) {
        $this->yellow = $yellow;
    }
    
    // Handle page content of shortcut
    public function onParseContentShortcut($page, $name, $text, $type) {
        $output = null;
        if ($name=="text" && ($type=="block" || $type=="inline")) {
            $output = $this->yellow->language->getTextHtml($text);
        }
        return $output;
    }
    
}
```

Name it `text.php`. Now you can write in your .md file something like this:

```
We would like to thank **very much** [text LanguageTranslator] for this beautiful translation!
```

**Reply by MiguelVis** [reply]
Thanks a lot @GiovanniSalmeri!

**Reply by markseuffert** [reply]
![meme-this-is-the-way](https://user-images.githubusercontent.com/2429165/136006429-2cf84dab-9707-4b0c-91f9-d9803973fecf.jpg)
